packages/@glimmer/integration-tests/test/compiler/compile-options-test.ts

```diff
@@ -72,6 +72,29 @@ module('[glimmer-compiler] precompile', ({ test }) => {
     assert.equal(componentName, 'rental', 'customized component name was used');
   });
 
+  test('customizeComponentName is not invoked on curly components', function (assert) {
+    let wire = JSON.parse(
+      precompile('{{#my-component}}hello{{/my-component}}', {
+        customizeComponentName(input: string) {
+          return input.toUpperCase();
+        },
+      })
+    );
+
+    let block: WireFormat.SerializedTemplateBlock = JSON.parse(wire.block);
+
+    let [[, componentNameExpr]] = block[0] as [WireFormat.Statements.Block];
+
+    glimmerAssert(
+      Array.isArray(componentNameExpr) &&
+        componentNameExpr[0] === SexpOpcodes.GetFreeAsComponentHead,
+      `component name is a free variable lookup`
+    );
+
+    let componentName = block[3][componentNameExpr[1]];
+    assert.equal(componentName, 'my-component', 'original component name was used');
+  });
```

Comment:
Can you make another (I think failing) test that looks like:

```hbs
{{#Foo}}hello{{/Foo}}
```

And also confirms that `customizeComponentName` is not called? I _believe_ that the current implementation will fail a test like this, but prior to #1253 would have passed.

Reply:
Looks like you're right: https://github.com/dfreeman/glimmer-vm/pull/1/checks?check_run_id=1772791358#step:5:296

Reply:
Done (thanks @jamescdavis!)